src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -1086,3 +1088,48 @@ def test_node_constructor_with_parent():
     assert node.has_constructor_parent is False
     wrong_parent.addchild(node.detach())
     assert node.parent is wrong_parent
+
+
+def test_following_preceding():
+    '''Test that the preceding and following() methods in the Node class
+    behave as expected.
+
+    '''
+    # 1: There is no Routine ancestor node.
+    a_ref = Reference(DataSymbol("a", REAL_TYPE))
+    b_ref = Reference(DataSymbol("b", REAL_TYPE))
+    c_ref = Reference(DataSymbol("c", REAL_TYPE))
+    d_ref = Reference(DataSymbol("d", REAL_TYPE))
+    multiply1 = BinaryOperation.create(
+        BinaryOperation.Operator.MUL, c_ref, d_ref)
+    multiply2 = BinaryOperation.create(
+        BinaryOperation.Operator.MUL, b_ref, multiply1)
+    assign = Assignment.create(a_ref, multiply2)
+
+    # 1a: First node.
+    assert assign.following() == [
+        a_ref, multiply2, b_ref, multiply1, c_ref, d_ref]
+    assert not assign.preceding()
+
+    # 1b: Last node.
+    assert not d_ref.following()
+    assert d_ref.preceding() == [
+        assign, a_ref, multiply2, b_ref, multiply1, c_ref]
+    assert d_ref.preceding(reverse=True) == [
```

Comment:
It's not the fault of this PR but it seems strange to me that we bother supporting the `reverse` flag when all that is needed is a `.reverse()` on the result.

Reply:
(You can ignore this though.)

Reply:
I agree.